schema_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provider
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/ec2"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUnderlyingType(t *testing.T) {
```

Comment:
I think this would really benefit from a technique called table based testing. We do this all over, but a good example can be found here: https://github.com/pulumi/pulumi/blob/7827eb26bc1a64aa56ad450d35bccde0ceadd78e/nil#L10-L51